index.js

```diff
@@ -1,16 +1,34 @@
-var express = require('express');
-var mongoose = require('mongoose');
-var bluebird = require('bluebird');
-mongoose.Promise = bluebird;
-
-var app = express();
+const express = require('express');
+const mongoose = require('mongoose');
+const bluebird = require('bluebird');
+const socketHandler = require('./socketHandler');
 
-var port = process.env.PORT || 3000;
+// App init
+const app = express();
 
-// configure our server with all the middleware and routing
+// Configure server with all the middleware and routing
 require('./middleware.js')(app, express);
 require('./routes.js')(app, express);
 
-app.listen(port, function() {
-  console.log('Listening on port ' + port);
+// Export App for testing
+module.exports = app;
+
+// Connect socket
+const server = require('http').Server(app);
+const io = require('socket.io')(server);
+mongoose.Promise = bluebird;
+
+// Connect to mongoDB unless testing
+const port = process.env.PORT || 3000;
+if (process.env.TEST === 'false') {
```

Comment:
I'd like to replace this with `process.env.NODE_ENV === 'test'` eventually. That would allow us to quickly specify any number of environment types -- not just `test`.